### PR TITLE
replace DocumentSource::NotFromParser with DocumentSource::FromParser

### PR DESCRIPTION
**Pull request by tigercosmos**
per https://github.com/servo/servo/pull/4297/files#r158729975

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19647)

### CODE REVIEW

File: components/script/dom/domparser.rs [hunk 1]
```diff
@@ -77,7 +77,6 @@ impl DOMParserMethods for DOMParser {
                 Ok(document)
             }
             Text_xml | Application_xml | Application_xhtml_xml => {
-                // FIXME: this should probably be FromParser when we actually parse the string (#3756).
```
Comment (emilio) [inline components/script/dom/domparser.rs]: This probably has some behavior change, and thus needs a test.

In particular, this determines the initial `ReadyState` and such, so we should be able to test whether parsing a document with a subresource claims to be loaded or not from the start.

Though @jdm or @nox probably know this code better.

Comment (tigercosmos) [reply components/script/dom/domparser.rs]: please let me know how to do to start to test. :)

Comment (jdm) [reply components/script/dom/domparser.rs]: https://github.com/servo/servo/blob/master/tests/wpt/README.md#writing-new-tests has instructions for creating new tests. For this one in particular, we want to test that the document returned from `(new DOMParser()).parseFromString("<html></html>", "text/xml")` has `readyState` equal to `complete`.

**Comment by tigercosmos** [comment]
`(new DOMParser()).parseFromString("<html></html>", "text/xml").readyState` will get `loading`

Something wrong need to check.

**Comment by jdm** [comment]
I think I see the issue - the Text_html branch has an explicit set_ready_state call.

**Comment by tigercosmos** [comment]
You're right. It misses argument to set readyState.

**Comment by tigercosmos** [comment]
Could someone take a look?

File: tests/wpt/mozilla/tests/mozilla/xmldomparser.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>XML Dom Parse readyState Test</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+    test(function () {
+        assert_equals(
+            (new DOMParser()).parseFromString("<html></html>", "text/xml").readyState,
```
Comment (emilio) [inline tests/wpt/mozilla/tests/mozilla/xmldomparser.html]: We should move it into `web-platform-tests` instead of `mozilla`.

**Review by emilio — COMMENTED** [review]
This looks reasonable to me, but @jdm should be the one stamping it :)

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Pull request merged**